src/i_video.c

```diff
@@ -2165,6 +2167,36 @@ const pixel_t I_BlendOverAltTinttab (const pixel_t bg, const pixel_t fg)
 	return amask | r | g | b;
 }
 
+// [crispy] More opaque (75%) XLATAB blending emulation, used for Strife
+const pixel_t I_BlendOverXlatab (const pixel_t bg, const pixel_t fg)
+{
+	const uint32_t r = ((blend_alpha_xlatab * (fg & rmask) + (0xff - blend_alpha_xlatab) * (bg & rmask)) >> 8) & rmask;
+	const uint32_t g = ((blend_alpha_xlatab * (fg & gmask) + (0xff - blend_alpha_xlatab) * (bg & gmask)) >> 8) & gmask;
+	const uint32_t b = ((blend_alpha_xlatab * (fg & bmask) + (0xff - blend_alpha_xlatab) * (bg & bmask)) >> 8) & bmask;
+
+	return amask | r | g | b;
+}
+
+// [crispy] Less opaque (25%) XLATAB blending emulation, used for Strife
+const pixel_t I_BlendOverAltXlatab (const pixel_t bg, const pixel_t fg)
+{
+	const uint32_t r = ((blend_alpha_altxlatab * (fg & rmask) + (0xff - blend_alpha_altxlatab) * (bg & rmask)) >> 8) & rmask;
+	const uint32_t g = ((blend_alpha_altxlatab * (fg & gmask) + (0xff - blend_alpha_altxlatab) * (bg & gmask)) >> 8) & gmask;
+	const uint32_t b = ((blend_alpha_altxlatab * (fg & bmask) + (0xff - blend_alpha_altxlatab) * (bg & bmask)) >> 8) & bmask;
+
+	return amask | r | g | b;
+}
+
+// [crispy] Perform screen crossfading effect by given amount of opacity, used for Strife
+const pixel_t I_BlendOverCrossfade (const pixel_t bg, const pixel_t fg, const int amount)
```

Comment:
I think now we have reached the point where we can leave this as the only actual implementation of `I_BlendOver()` and have all the other variants turned into macros with hard-coded `amount` values.

Reply:
I've read all of your remarks, but let's please start from this one. I've got your idea of simplification and reducing amount of existing duplications and opacity values. This should be something like this:

<details>

```
const pixel_t I_BlendOver (const pixel_t bg, const pixel_t fg, const int amount)
{
	const uint32_t r = ((amount * (fg & rmask) + (0xff - amount) * (bg & rmask)) >> 8) & rmask;
	const uint32_t g = ((amount * (fg & gmask) + (0xff - amount) * (bg & gmask)) >> 8) & gmask;
	const uint32_t b = ((amount * (fg & bmask) + (0xff - amount) * (bg & bmask)) >> 8) & bmask;

	return amask | r | g | b;
}

// This should replace Doom's I_BlendOver calls:
const pixel_t I_BlendOverTranmap (const pixel_t bg, const pixel_t fg)
{
    // ... i.e. when I_BlendOverTranmap is called, catch it's values and pass it to main BlendOver function
    return I_BlendOver(bg, fg, 0xa8); // 168
}
```

</details>

But... You mean exactly to macrocize such "passing" `pixel_t` functions? Is it possible at all, without writing them in .c files? I never tried to write whole function as macro, so small hint or example will be very appreciated.

Reply:
Something like
```C
#define I_BlendOverTranmap(a,b) I_BlendOver(a,b,0xa8)
```
with the corresponding comment in `v_video.h`.

Reply:
Ah, dammit, function pointers. Right this won't work. The solution you suggested before looked reasonable, though.

Reply:
Done in https://github.com/fabiangreffrath/crispy-doom/pull/1183/commits/999da5c97abb2b56ed5193e7bd9fc33b03e2a6b3. Yeah, to make pointers and cases [like this](https://github.com/fabiangreffrath/crispy-doom/blob/master/src/doom/r_things.c#L820) happy, my another idea was to extend `I_BlendAdd` with incoming `amount` parameter, but this is doesn't reasonable, since additive blending doesn't have any meanings of "amount" at all, and such parameter will be simply useless.

Reply:
Thank you!